### PR TITLE
[polaris.shopify.com][/foundations][/patterns] Locations > Resource picking

### PR DESCRIPTION
**Pull request by chloerice**
### WHY are these changes introduced?

Follow up to #6596

### WHAT is this pull request doing?

- Updates the `/foundations/patterns/locations` Locations  pattern doc to document Pickers with Locations and Customer segments as examples.
- Adds storybook and style guide examples for searchable listbox to `Listbox`, `Popover`, and `Sheet`

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by martenbjork — APPROVED** [review]

File: polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+import {useUniqueId} from '../../../../utilities/unique-id';
+
 export interface PolarisContainerProps {}
 
 function PortalsContainerComponent(
   _props: PolarisContainerProps,
   ref: RefObject<HTMLDivElement>,
 ) {
-  return <div id="PolarisPortalsContainer" ref={ref} />;
+  const uniqueId = useUniqueId('PortalsContainer');
```
Comment (chloerice) [inline polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx]: (Fixes duplicate ID pa11y CI errors)

Comment (alex-page) [reply polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx]: This should be resolved with a merge from `main`. This happened when two `AppProviders` wrapped a component as a bug caused by me 😭

Comment (chloerice) [reply polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx]: Ahh okay thank you!! This CI check was driving me nuts 😁

Comment (alex-page) [reply polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx]: My bad Chloe.

Comment (chloerice) [reply polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx]: No worries!!!

Comment (chloerice) [reply polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx]: Hmmm the saga continues lol, CI is still complaining about duplicate IDs for the PortalsContainer, even though only one example and only one PortalsContainer ever gets rendered at a time 🧐

Comment (alex-page) [reply polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx]: It looks good now. When there are zero changes in `polaris-react` the a11y test doesn't run.

Comment (alex-page) [reply polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx]: https://github.com/Shopify/polaris/blob/fdf4d836dccedd4578d094fa216c5460780064cc/.github/workflows/ci-a11y-vrt.yml#L3-L16

Comment (alex-page) [reply polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx]: Ok I am silly. Fixed here 31b860374fc101c55792f7dbfb0ffc4f67179ed1

Comment (alex-page) [reply polaris-react/src/components/PortalsManager/components/PortalsContainer/PortalsContainer.tsx]: Sorry for the inconvenience Chloe 😬

**Pull request merged**